### PR TITLE
Anion's `hdd_usage` Segment

### PR DESCRIPTION
**Pull request by bhilburn**
Hey @anion155! @dritter spotted your custom `hdd_usage` segment in your fork, and we think this would be a great candidate to upstream! Creating this PR to get the discussion going.

This PR is currently based on `master`, but needs to be rebased against `next`.

### CODE REVIEW

File: functions/icons.zsh [hunk 1]
```diff
@@ -34,6 +34,7 @@ case $POWERLEVEL9K_MODE in
       TEST_ICON                      $'\uE891'              # 
       TODO_ICON                      $'\u2611'              # ☑
       BATTERY_ICON                   $'\uE894'              # 
+      HDD_ICON                       $'\uD83D\uDDB4'        # 🖴
```
Comment (dritter) [inline functions/icons.zsh]: I'm using `awesome-patched` mode and cannot see this icon...
How about `$'\uE1AE'`?

File: functions/icons.zsh [hunk 2]
```diff
@@ -92,8 +93,9 @@ case $POWERLEVEL9K_MODE in
       TEST_ICON                      $'\uF291'              # 
       TODO_ICON                      $'\u2611'              # ☑
       BATTERY_ICON                   $'\U1F50B'             # 🔋
-      OK_ICON                        $'\uF23A'              # 
-      FAIL_ICON                      $'\uF281'              # 
+      HDD_ICON                       $'\uF0A0 '             # 
+      OK_ICON                        $'\uF00C '             # 
```
Comment (dritter) [inline functions/icons.zsh]: I would prefer not to change codepoints for unrelated Icons. If there are better icons, they should be updated in a separate pull request.

File: powerlevel9k.zsh-theme [hunk 3]
```diff
@@ -337,6 +337,55 @@ prompt_background_jobs() {
   fi
 }
 
+# Segment to indicate hdd available level.
+set_default POWERLEVEL9K_HDD_USAGE_ONLY_WARNING false
+set_default POWERLEVEL9K_HDD_USAGE_WARNING_LEVEL 10
+set_default POWERLEVEL9K_HDD_USAGE_CRITICAL_LEVEL 5
+prompt_hdd_usage() {
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: You named the segment `hdd_usage`, but it actually shows the amount of free space, right? I think another name would make that clearer.
Otherwise it could be nice to add an option, so that the user can decide what to show free space or usage..

File: powerlevel9k.zsh-theme [hunk 4]
```diff
@@ -399,8 +448,8 @@ prompt_battery() {
     [[ -z $bat ]] && return
 
     [[ $(cat $bat/capacity) -gt 100 ]] && local bat_percent=100 || local bat_percent=$(cat $bat/capacity)
-    [[ $(cat $bat/status) =~ Charging ]] && local connected=true
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: Unrelated to hdd_usage..

Comment (bhilburn) [inline powerlevel9k.zsh-theme]: Is this related to #348, I wonder?

File: powerlevel9k.zsh-theme (hunk 3 above)
Comment (bhilburn) [inline powerlevel9k.zsh-theme]: Great call - I imagine people will want both options.

File: functions/icons.zsh (hunk 2 above)
Comment (bhilburn) [inline functions/icons.zsh]: Agreed with @dritter on this one.

**Comment by bhilburn** [comment]
No comments from @anion155, yet. I would really like to get his input on moving forward, if at all possible.

File: functions/icons.zsh (hunk 1 above)
Comment (bhilburn) [inline functions/icons.zsh]: Fixed this to be consistent in new branch.

File: functions/icons.zsh (hunk 2 above)
Comment (bhilburn) [inline functions/icons.zsh]: Removed.

File: powerlevel9k.zsh-theme (hunk 3 above)
Comment (bhilburn) [inline powerlevel9k.zsh-theme]: So, actually, this code does implement usage. It's using `df`, which by default will give you usage percent. It also gives you "available blocks", but that's not what the math is based on. Column 5 in the below: 
usage

File: powerlevel9k.zsh-theme (hunk 4 above)
Comment (bhilburn) [inline powerlevel9k.zsh-theme]: Done.

**Comment by bhilburn** [comment]
No response from @anion155, so I created a new branch from `next`, cherry-picked his commits that create the HDD segment (but leave out the other changes), made modifications per @dritter's review, and added some other things (README changes, etc.,). Posting a new PR shortly...

**Comment by bhilburn** [comment]
BTW, you were right, despite what I said in my review response - it *is* showing the available. There's a mix of comments / code that are somewhat contradictory.
